src/Translate/TranslateClient.php

```diff
@@ -183,24 +190,34 @@ public function translate($string, array $options = [])
      */
     public function translateBatch(array $strings, array $options = [])
     {
+        $options += [
+            'model' => '',
+        ];
+
         $response = $this->connection->listTranslations($options + [
             'q' => $strings,
             'key' => $this->key,
-            'target' => $this->targetLanguage
+            'target' => $this->targetLanguage,
+            'model' => $options['model']
         ]);
 
         $translations = [];
 
-        foreach ($response['data']['translations'] as $key => $translation) {
-            $source = isset($translation['detectedSourceLanguage'])
-                ? $translation['detectedSourceLanguage']
-                : $options['source'];
+        if (isset($response['data']['translations'])) {
+            foreach ($response['data']['translations'] as $key => $translation) {
+                $source = isset($translation['detectedSourceLanguage'])
+                    ? $translation['detectedSourceLanguage']
+                    : $options['source'];
 
-            $translations[] = [
-                'source' => $source,
-                'input' => $strings[$key],
-                'text' => $translation['translatedText']
-            ];
+                $translations[] = [
+                    'source' => $source,
+                    'input' => $strings[$key],
+                    'text' => $translation['translatedText'],
+                    'model' => (isset($translation['model']))
```

Comment:
Small code style nit - if we have the ternary for `$source` above do we need the one for model nested in the array definition here? I don't have a huge preference either way, but having them be formatted similarly would make sense to me.